### PR TITLE
C sample code.

### LINKED ISSUE #4: Add example directory with testing C code
Sample C code, not necessarily from libamtrack. Code should contain proper doxygen and be structured in header and source files.

Ideally it should cover all cases:
 - only in parameters
 - only out parameters
 - mixed in, out, in/out
 - multiple out parameters
 - mutplitle out parameters with different multiplicites
 - functions with no parameters

### PR DESCRIPTION
**Pull request by Sharon131**
Fixes #4

### CODE REVIEW

**Comment by ptyszko** [comment]
I think testing would be more productive if the functions actually assigned something to the output parameters

File: C_samples/multiple_in_out_params_sample.c [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+#include <stdbool.h>
+
+#include "multiple_in_out_params_sample.h"
+
+
+/* -------------------- One in, two out params ------------------------*/
+
+void one_in_two_out_params_no_return_function(int param_in, double* param_out_one, char* param_out_two)
+{
+    printf("\r\n");
```
Comment (grzanka) [inline C_samples/multiple_in_out_params_sample.c:11]: Calculate some values of output parameters, i.e. `*param_out_one = param_in * 2.0`

Comment (grzanka) [reply C_samples/multiple_in_out_params_sample.c:11]: Same in other cases

File: C_samples/multiple_in_out_params_sample.c [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+int one_in_two_out_params_int_return_function(int param_in, double* param_out_one, char* param_out_two)
+{
+    return -1;
```
Comment (grzanka) [inline C_samples/multiple_in_out_params_sample.c:16]: return some values which depend on input params

**Review by grzanka — CHANGES_REQUESTED** [review]
Add support for tables (of doubles, floats and ints) of variable size, as input and output of functions.
There may be multiple tables as input and output

**Comment by grzanka** [comment]
Add some code (i.e. `main.c`) which calls tests functions

**Review by grzanka — APPROVED** [review]
Conditionally approving, separate issues created

**Pull request merged**